src/vs/workbench/contrib/interactiveEditor/browser/interactiveEditorActions.ts

```diff
@@ -415,14 +408,39 @@ export class ApplyPreviewEdits extends AbstractInteractiveEditorAction {
 			logService.warn('FAILED to apply changes, no edit response');
 			return;
 		}
-		ctrl.cancelSession();
 		if (edit.singleCreateFileEdit) {
 			editorService.openEditor({ resource: edit.singleCreateFileEdit.uri }, SIDE_GROUP);
 		}
 
 	}
 }
 
+export class CancelSessionAction extends AbstractInteractiveEditorAction {
+
+	constructor() {
+		super({
+			id: 'interactiveEditor.cancel',
+			title: localize('discard', 'Discard Changes'),
+			icon: Codicon.clearAll,
+			precondition: CTX_INTERACTIVE_EDITOR_VISIBLE,
+			keybinding: {
+				weight: KeybindingWeight.EditorContrib - 1,
+				primary: KeyCode.Escape
+			},
+			menu: {
+				id: MENU_INTERACTIVE_EDITOR_WIDGET_STATUS,
+				// when: CTX_INTERACTIVE_EDITOR_EDIT_MODE.isEqualTo('preview'),
```

Comment:
Intentional comment?

Reply:
Yeah, for now I want to enable this for all modes